cpp/ql/lib/semmle/code/cpp/controlflow/IRGuards.qll

```diff
@@ -1157,4 +1157,4 @@ private predicate add_eq(
 }
 
 /** The int value of integer constant expression. */
-private int int_value(Instruction i) { result = i.(IntegerConstantInstruction).getValue().toInt() }
+private int int_value(Instruction i) { result = i.(ConstantInstruction).getValue().toInt() }
```

Comment:
Wouldn't this mean that comparisons with the float `0.0` would now also be ok?

Reply:
Hmm, good question. I would hope that `toInt` wouldn't have a result for that, but that's definitly worth a test. I guess we could place a restriction on the instruction saying that it should either be a `IntegerConstantInstruction` (which would cover `int` and `bool`) or a `PointerConstantInstruction` (or whatever we may call a constant instruction with a pointer type.

Reply:
Well spotted! That was indeed a problem. I've force-pushed a new commit history to highlight the problem better more clearly, and also show that the fix actually prevents this problem.